frontend/src/utils/axiosHeaders.js

```diff
@@ -0,0 +1,24 @@
+import authHeader from '../services/auth-header'
+
+export const validHeader = (params) => ({
+  headers: {
+    ...authHeader()
+  },
+  params: params
+})
+
+export const multipartFileHeader = (params) => ({
+  headers: {
+    ...authHeader(),
+    'Content-Type': 'multipart/form-data'
+  },
+  params: params
+})
+
+export const fileHeaderWithParams = (params) => ({
+  headers: {
+    ...authHeader()
+  },
+  responseType: 'blob',
+  params: params
+})
```

Comment:
Ogólnie nie mam już żadnych zastrzeżeń, ale chciałbym Ci tylko zaproponować małą zmianę:
```suggestion
export const validHeader = (params, optionalData = {}) => ({
  headers: {
    ...authHeader(),
    'Content-Type': optionalData?.contentType
  },
  responseType: optionalData?.responseType,
  params: params
})

export const multipartFileHeader = (params) => validHeader(params, { contentType: 'multipart/form-data' })
export const blobResponseHeader = (params) => validHeader(params, { responseType: 'blob' })
```

Po pierwsze sprawdź wgl czy to ma prawo działać i czy zadziała dla każdego przypadku użycia. Jeżeli tak to rozważ takie rozwiązanie bo było by chyba bardziej elegancko (?)

Reply:
Odnośnie paramsów - myślę że deafult value to dobry pomysł, sam osobiście zostawiłbym na razie wydzielone headery, moim zdaniem taka parametryzacja jest trochę brzydka i zmniejsza czytelność kodu

\+ każda zmiana/naprawa jak coś się sypnie jest łatwiejsza do naprawienia kiedy te rzeczy są na razie podzielone, nawet jeśli DRY trochę cierpi

Reply:
Spory problem ze znalezieniem problemów z headerem wcześniej był właśnie przez takie składanie tego i trochę niejasną strukturę, więc może po prostu mam uraz ale szczerze to raczej bym zostawił jak jest